tests/engine/results/test_results.py

```diff
@@ -204,3 +204,17 @@ def test_local_exists_full_path(self, tmp_dir):
     def test_local_init_with_different_drive_works_on_windows(self):
         result = LocalResult(dir="E:/location", validate_dir=False)
         assert result.dir == "E:/location"
+
+
+class TestResultFormatting:
+    def test_result_accepts_callable_for_location(self, tmpdir):
+        result = LocalResult(dir=tmpdir, location=lambda **kwargs: "special_val")
+        assert result.location is None
+        new_result = result.format()
+        assert new_result.location.endswith("special_val")
+
+    def test_result_callable_accepts_kwargs(self, tmpdir):
+        result = LocalResult(dir=tmpdir, location=lambda **kwargs: kwargs["key"])
```

Comment:
This is pretty neat 🙂

Reply:
Indeed!
I'd replace the lambda function with `operator.itemgetter("key")`